mdsuite/transformations/thermal_flux.py

```diff
@@ -24,193 +24,71 @@
-            pe_path = str.encode(join_path(species, "PE"))
+        super(ThermalFlux, self).__init__(
+            input_properties=[
+                mdsuite_properties.stress,
+                mdsuite_properties.velocities,
+                mdsuite_properties.kinetic_energy,
+                mdsuite_properties.potential_energy,
+            ],
+            output_property=mdsuite_properties.thermal_flux,
+            batchable_axes=[1, 2],
+            scale_function={"linear": {"scale_factor": 5}},
+        )
+
+    def transform_batch(
+        self, batch: typing.Dict[str, typing.Dict[str, tf.Tensor]]
+    ) -> tf.Tensor:
+        fluxes = []
+        for _, properties in batch.items():
+            stress = properties[mdsuite_properties.stress.name]
+            vel = properties[mdsuite_properties.velocities.name]
+            ke = properties[mdsuite_properties.kinetic_energy.name]
+            pe = properties[mdsuite_properties.potential_energy.name]
             phi_x = (
-                np.multiply(data[stress_path][:, :, 0], data[velocity_path][:, :, 0])
-                + np.multiply(data[stress_path][:, :, 3], data[velocity_path][:, :, 1])
-                + np.multiply(data[stress_path][:, :, 4], data[velocity_path][:, :, 2])
+                np.multiply(stress[:, :, 0], vel[:, :, 0])
+                + np.multiply(stress[:, :, 3], vel[:, :, 1])
+                + np.multiply(stress[:, :, 4], vel[:, :, 2])
             )
             phi_y = (
-                np.multiply(data[stress_path][:, :, 3], data[velocity_path][:, :, 0])
-                + np.multiply(data[stress_path][:, :, 1], data[velocity_path][:, :, 1])
-                + np.multiply(data[stress_path][:, :, 5], data[velocity_path][:, :, 2])
+                np.multiply(stress[:, :, 3], vel[:, :, 0])
+                + np.multiply(stress[:, :, 1], vel[:, :, 1])
+                + np.multiply(stress[:, :, 5], vel[:, :, 2])
             )
             phi_z = (
-                np.multiply(data[stress_path][:, :, 4], data[velocity_path][:, :, 0])
-                + np.multiply(data[stress_path][:, :, 5], data[velocity_path][:, :, 1])
-                + np.multiply(data[stress_path][:, :, 2], data[velocity_path][:, :, 2])
+                np.multiply(stress[:, :, 4], vel[:, :, 0])
+                + np.multiply(stress[:, :, 5], vel[:, :, 1])
+                + np.multiply(stress[:, :, 2], vel[:, :, 2])
             )
 
             phi = np.dstack([phi_x, phi_y, phi_z])
 
-            phi_sum = phi.sum(axis=0)
-            phi_sum_atoms = (
-                phi_sum / self.experiment.units["NkTV2p"]
-            )  # factor for units lammps nktv2p
+            phi_sum_atoms = phi.sum(axis=0)
+            # phi_sum_atoms = (
+            #         phi_sum_atoms / self.experiment.units["NkTV2p"]
+            # )  # factor for units lammps nktv2p
+            # TODO why is there a unit conversion in the transformation?
```

Comment:
This should be adressed! @SamTov do you know this?

Reply:
It is more a scaling factor from memory not a direct unit conversion. It should retain the units. But this is from @Fratorhe so he should confirm that.